src/components/events/EditModal.jsx

```diff
@@ -0,0 +1,187 @@
+                setError('Please fill fields.')
+                return
+            }
+            await axios.put(`${process.env.REACT_APP_API_URL}/api/deal/edit/${_id}`, {
+                ...newDeal
+            }, {
+                headers: {
+                  Authorization: `${user.token}`,
+                },
+              })
+              window.location.reload()
+        } catch(e) {
+            setError(e)
+            console.log(e)
+        }
+    }
+
+    const onClickDelete = async () => {
+        try {
+            await DeleteDeal(_id, user);
+            window.location.reload()
+        } catch(e) {
+            console.log(e)
+        }
+    }
+    return (
+        <div className="fixed top-0 left-0 w-full h-full z-50">
+            <div
+                className="absolute top-0 left-0 h-full w-full bg-gray-500 opacity-50 z-30 overflow-hidden"
+                onClick={() => onDismiss()}
+            />
+            {/* <!-- Modal content --> */}
+            <div className="flex justify-center ">
+                <div
+                    id="reportModal"
+                    tabIndex="-1"
+                    aria-hidden="true"
+                    className={`fixed top-1/4 mx-auto h-fit z-50 flex w-fit p-4 overflow-hidden`}
+                >
+                    <div className="bg-white rounded-lg shadow w-96">
+                        {/* <!-- Modal header --> */}
+                        <div className="flex items-start justify-between p-4 border-b rounded-t">
+                            <h3 className="text-xl font-semibold text-gray-900 pt-1">
+                                Edit Deal
+                            </h3>
+                            <ErrorText error={error} />
+                            <button
+                                type="button"
+                                className="text-gray-400 bg-transparent rounded-lg text-sm p-1.5 ml-auto inline-flex items-center"
+                                onClick={() => onDismiss()}
+                            >
+                                <XIcon />
+                                <span className="sr-only">Close modal</span>
+                            </button>
+                        </div>
+                        {/* <!-- Modal body --> */}
+                        <div className="p-6 space-y-2">
+                            <GenericInput
+                                name="Title"
+                                type="text"
+                                label="Title"
+                                extraProps="w-full"
+                                required
+                                placeholder="Taco Tuesday"
+                                onChange={(e) => setNewDeal({ ...newDeal, title: e.target.value })}
+                                icon={<IdIcon />}
+                                value={newDeal.title || title}
+                            />
+                            <GenericInput
```

Comment:
This should not be a generic input element. This needs to utilize the same input as used in add deal: https://github.com/alexsaltstein/HobokenEventsWebApp/blob/dev/src/components/form/AddDealElement.js#L189 type="time" Also we may want to make this time element a generic now that we use it in multiple spots